### PR TITLE
Rewrite logic to determine if a show is live as well as embed logic

### PR DESCRIPTION
**Pull request by BrianMMcClain**

### CODE REVIEW

File: themes/docsy/layouts/partials/head-scripts.html [hunk 1]
```diff
@@ -100,37 +100,39 @@
...
+  currentTime = new Date(Date.now());
+  for (var episode of episodes) {
+    if (currentTime >= episode["startTime"] && currentTime <= episode["endTime"]) {      
+      // If a specific show was defined, make sure it matches
+      if (showName == null || showName == episode["show"]) {
+        // Live show found
+        liveShowName = episode["show"];
+        liveShowLink = episode["link"];
+        liveShowStream = episode["stream"];
+        return true;
```
Comment (bryanfriedman) [inline themes/docsy/layouts/partials/head-scripts.html:129]: Should we set a variable and return rather than having two different return statements?

**Review by bryanfriedman — COMMENTED** [review]
Nice, this is way simpler. JavaScript objects ftw. Just a couple comments but this is a huge improvement.

Is this using UTC instead of timezones, or is it doing the calculations with the local timezone?

File: themes/docsy/layouts/partials/head-scripts.html [hunk 2]
```diff
@@ -100,37 +100,39 @@
...
 function isTvShowLive(showName) {   
+  // Gather all the episodes that have a start time after one week ago
+  // This is due to the Go template being gathered at build time, not
+  // in real time
+  episodes = [];
+  {{ range where (where .Site.Pages "Type" "tv-episode") "Date.Unix" "ge" (sub now.Unix 604800) }}
+    var epInfo = {
+      startTime: new Date("{{ .Params.Date }}"),
+      endTime: new Date(new Date("{{ .Params.Date }}").getTime() + ({{ default 60 .Params.minutes }} * 60000)),
+      duration: {{ default 60 .Params.minutes }} * 60000,
```
[the remark this answers is not in the document]
Comment (BrianMMcClain) [reply themes/docsy/layouts/partials/head-scripts.html:111]: Good catch, we don't anymore. I had this used in a different piece of logic at one point but it's unreferenced now. I'll remove it in the next commit.

File: themes/docsy/layouts/partials/head-scripts.html (hunk 1 above)
Comment (BrianMMcClain) [reply themes/docsy/layouts/partials/head-scripts.html:129]: That works as well, I'll shuffle it around in the next commit

**Comment by BrianMMcClain** [comment]
> Is this using UTC instead of timezones, or is it doing the calculations with the local timezone?

Well, I did some more reading around this morning so I think I need to do some verification just to be safe. My initial reading yesterday had led me to believe that was the case, but there's a nuance to it that I came across in the [MDN documentation](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date):

> Note: It's important to keep in mind that while the time value at the heart of a Date object is UTC, the basic methods to fetch the date and time or its components all work in the local (i.e. host system) time zone and offset. We only really use two pieces of the `Date` object: the constructor and the `Date.now()` function.

Looking at the `Date.now()` function, the docs read:

> Returns the numeric value corresponding to the current time—the number of milliseconds elapsed since January 1, 1970 00:00:00 UTC, with leap seconds ignored.

So as I read that, that should operate as UTC all the way through 100%. The constructor is where it gets a bit murky. We use two versions of the constructor: one with milliseconds passed, and one with a datetime string passed:

For the one with ms passed (ie. `new Date(Date.now())`), we should be good there as well:

>2. Time value or timestamp number
> value
> An integer value representing the number of milliseconds since January 1, 1970, 00:00:00 UTC (the ECMAScript epoch, equivalent to the UNIX epoch), with leap seconds ignored. Keep in mind that most UNIX Timestamp functions are only accurate to the nearest second.

Passing the Date from the show metadata invokes a different constuctor:

> 3. Timestamp string
> dateString
> A string value representing a date, specified in a format recognized by the Date.parse() method. (These formats are IETF-compliant RFC 2822 timestamps, and also strings in a version of ISO8601.)
>
>Note: Parsing of date strings with the Date constructor (and Date.parse(), which works the same way) is strongly discouraged due to browser differences and inconsistencies.
>
> - Support for RFC 2822 format strings is by convention only.
> - Support for ISO 8601 formats differs in that date-only strings (e.g. "1970-01-01") are treated as UTC, not local.

So two points there, first just the recommendation to just not do that, which is echoed in the the [`Date.parse() docs`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/parse):

> It is not recommended to use Date.parse as until ES5, parsing of strings was entirely implementation dependent. There are still many differences in how different hosts parse date strings, therefore date strings should be manually parsed (a library can help if many different formats are to be accommodated).

However, if we believe that's ok, it looks like it does operate in UTC as well:

> The Date.parse() method parses a string representation of a date, and returns the number of milliseconds since January 1, 1970, 00:00:00 UTC or NaN if the string is unrecognized or, in some cases, contains illegal date values (e.g. 2015-02-31).

> While time zone specifiers are used during date string parsing to interpret the argument, the value returned is always the number of milliseconds between January 1, 1970 00:00:00 UTC and the point in time represented by the argument or NaN.



tl;dr: Yeah UTC :)

**Comment by BrianMMcClain** [comment]
@bryanfriedman I've also added a commit that embeds the Twitch stream if the episode is live. This will need more nuance written around it if we explore other channels however.

**Comment by BrianMMcClain** [comment]
@bryanfriedman @tiffanyfay with the addition of embedding live shows on the episode page, do we want the top banner notifying the reader that a show is live to direct them to the episode page? Or keep it directing them to the show page?

**Comment by tiffanyfay** [comment]
> @bryanfriedman @tiffanyfay with the addition of embedding live shows on the episode page, do we want the top banner notifying the reader that a show is live to direct them to the episode page? Or keep it directing them to the show page?

I think it should go to the episode page.
